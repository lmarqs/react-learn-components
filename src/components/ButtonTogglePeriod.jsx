import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { toogleDayNight } from '../actions/AppActions';

import Image from '../day-and-night.png';

const ButtonTogglePeriod = ({ isNight, toogleDayNight }) => {
  const pos = (isNight) ? "active" : "";
  return (
    <button
      onClick={toogleDayNight}
      className={ "toggle-button floating " + pos }>
      <img src={Image} alt="Toggle day/night" width="32" height="32" />
    </button>
  )
}
ButtonTogglePeriod.propTypes = {
  isNight: PropTypes.bool,
  toogleDayNight: PropTypes.func,
}

export const mapStateToProps = ({ app }) => {
  const { isNight } = app
  return { isNight }
}

export default connect(mapStateToProps, { toogleDayNight })(ButtonTogglePeriod)
