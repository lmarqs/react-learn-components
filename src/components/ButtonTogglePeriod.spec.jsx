import React from 'react';
import renderer from 'react-test-renderer';
import ButtonTogglePeriod from './ButtonTogglePeriod';

it('renders correctly', () => {
    const component = renderer.create(<ButtonTogglePeriod onChangeEvent={jest.fn()} />)
    expect(component.toJSON()).toMatchSnapshot();
});

it('toggle active class after a click', () => {
    const component= renderer.create(<ButtonTogglePeriod onChangeEvent={jest.fn()} />)

    const button = component.root.findByType('button');

    renderer.act(() => {
        button.props.onClick();
    });

    expect(component.toJSON()).toMatchSnapshot();
});