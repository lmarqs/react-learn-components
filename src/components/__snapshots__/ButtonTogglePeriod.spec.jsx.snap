// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="toggle-button floating "
  onClick={[Function]}
>
  <img
    alt="Toggle day/night"
    height="32"
    src="day-and-night.png"
    width="32"
  />
</button>
`;

exports[`toggle active class after a click 1`] = `
<button
  className="toggle-button floating active"
  onClick={[Function]}
>
  <img
    alt="Toggle day/night"
    height="32"
    src="day-and-night.png"
    width="32"
  />
</button>
`;
